### PR TITLE
Evaluation code refactoring

### LINKED ISSUE #40: Test selector imports `ntpath` module
...and documentation clearly says:

```
DESCRIPTION
    Instead of importing this module directly, import os and refer to this
    module as os.path.
```

### LINKED ISSUE #41: The code uses undefined function
```
$ git grep -B5 file_diff_data_branch
tests_selector/evaluation/start.py-from tests_selector.utils.git import (
tests_selector/evaluation/start.py-    get_git_repo,
tests_selector/evaluation/start.py-    file_diff_data_current,
tests_selector/evaluation/start.py-    get_test_lines_and_update_lines,
tests_selector/evaluation/start.py-    changed_files_branch,
tests_selector/evaluation/start.py:    file_diff_data_branch,
--
tests_selector/evaluation/start.py-
tests_selector/evaluation/start.py-        changed_files = changed_files_branch(PROJECT_FOLDER)
tests_selector/evaluation/start.py-        tests_line_level = set()
tests_selector/evaluation/start.py-        tests_file_level = set()
tests_selector/evaluation/start.py-        for f in changed_files:
tests_selector/evaluation/start.py:            diff = file_diff_data_branch(f, PROJECT_FOLDER)
```

It's not defined anywhere, as far as I can tell

### PR DESCRIPTION
**Pull request by guotin**
closes #41 
closes #40 

* Changed evaluation code to `/tests/` folder as discussed in #45 
* Cleaned up useless code as much as possible and extracted evaluation code from source code
* Current status of evaluation code: 
  * running `tests_selector_eval` in target project repo removes `n` number of lines from a random target project source code file. Then it calculates line-level, file-level test suites and compares `pytest` exitcodes (saves them to `"results.db"`)
  * I guess plans on how to use this with testing are more clear in the future

### CODE REVIEW

File: tests_selector/tests/evaluation/eval_helpers.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    full_test_suite_size,
+    line_test_suite_size,
+    file_test_suite_size,
+    exitcode_line,
+    exitcode_file,
+    exitcode_all,
+    db_name,
+):
+    """Print random remove test statistics"""
+    print("============")
```
Comment (matveypashkovskiy) [inline tests_selector/tests/evaluation/eval_helpers.py:108]: Shall we start using `logging`? Something like:

```python
import logging

logger = logging.getLogger()
logger.info(message)
```

File: tests_selector/tests/evaluation/eval_helpers.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
+"""Helper functions for evaluation code"""
+import subprocess
+import random
+from tests_selector.utils.git import (
+    get_git_repo,
+)
+
+
+def checkout_remove_branch():
```
Comment (matveypashkovskiy) [inline tests_selector/tests/evaluation/eval_helpers.py:9]: Looks very similar to `TestHelper.checkout_new_branch()`, shouldn't it be used here with branch name as a parameter?

File: tests_selector/tests/evaluation/eval_helpers.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+
+def delete_random_lines_and_commit(deletes, src_files):
+    """Delete as many random lines as wanted from the list of given src-code files"""
+    repo = get_git_repo(None).repo.git
+    for i in range(deletes):
+        random_file = select_random_file(src_files)
+        filename = random_file[1]
+        delete_random_line(filename)
+        repo.add(filename)
```
Comment (matveypashkovskiy) [inline tests_selector/tests/evaluation/eval_helpers.py:60]: `TestHelper.commit_change()`?

**Comment by guotin** [comment]
Also just noticed that evaluation code doesn't work with `python 3.6.6` and working on fixing this

**Pull request merged**